[NativeAOT] System.Security.Tests failing on macOS with link errors

Failing in `/azp run runtime-extra-platforms` set with
```
  Undefined symbols for architecture arm64:
    "enum case for Foundation.Data.Deallocator.none(Foundation.Data.Deallocator.Type) -> Foundation.Data.Deallocator", referenced from:
        pal_swiftbindings.AppleCryptoNative_ChaCha20Poly1305Encrypt(keyPtr: Swift.UnsafeMutableRawPointer, keyLength: Swift.Int32, noncePtr: Swift.UnsafeMutableRawPointer, nonceLength: Swift.Int32, plaintextPtr: Swift.UnsafeMutableRawPointer, plaintextLength: Swift.Int32, ciphertextBuffer: Swift.UnsafeMutablePointer<Swift.UInt8>, ciphertextBufferLength: Swift.Int32, tagBuffer: Swift.UnsafeMutablePointer<Swift.UInt8>, tagBufferLength: Swift.Int32, aadPtr: Swift.UnsafeMutableRawPointer, aadLength: Swift.Int32) -> Swift.Int32 in libSystem.Security.Cryptography.Native.Apple.a(pal_swiftbindings.o)
        pal_swiftbindings.AppleCryptoNative_ChaCha20Poly1305Decrypt(keyPtr: Swift.UnsafeMutableRawPointer, keyLength: Swift.Int32, noncePtr: Swift.UnsafeMutableRawPointer, nonceLength: Swift.Int32, ciphertextPtr: Swift.UnsafeMutableRawPointer, ciphertextLength: Swift.Int32, tagPtr: Swift.UnsafeMutableRawPointer, tagLength: Swift.Int32, plaintextBuffer: Swift.UnsafeMutablePointer<Swift.UInt8>, plaintextBufferLength: Swift.Int32, aadPtr: Swift.UnsafeMutableRawPointer, aadLength: Swift.Int32) -> Swift.Int32 in libSystem.Security.Cryptography.Native.Apple.a(pal_swiftbindings.o)
```

```json
 {
    "ErrorMessage" : "pal_swiftbindings.AppleCryptoNative_ChaCha20Poly1305Encrypt",
    "BuildRetry": false
 }
```

Example of full log: https://dev.azure.com/dnceng-public/public/_build/results?buildId=37715&view=logs&jobId=77206cf6-10ff-51c9-d7c7-b6c46c149890&j=77206cf6-10ff-51c9-d7c7-b6c46c149890&t=004edc8c-fef8-5b51-d46d-fd60518d1541


<!--Known issue error report start -->
### Report
|Build|Definition|Step Name|Console log|
|---|---|---|---|
|[37715](https://dev.azure.com/dnceng-public/public/_build/results?buildId=37715)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/37715/logs/510)
|[37699](https://dev.azure.com/dnceng-public/public/_build/results?buildId=37699)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/37699/logs/561)
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|2|2|2|
<!--Known issue error report end -->

--- a/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets
+++ b/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets
@@ -76,6 +76,7 @@ The .NET Foundation licenses this file to you under the MIT license.
 
     <ItemGroup Condition="'$(TargetOS)' == 'OSX'">
       <NativeFramework Include="CoreFoundation" />
+      <NativeFramework Include="CryptoKit" />
       <NativeFramework Include="Foundation" />
       <NativeFramework Include="Security" />
       <NativeFramework Include="GSS" />
@@ -98,9 +99,13 @@ The .NET Foundation licenses this file to you under the MIT license.
       <LinkerArg Include="-lstdc++" />
       <LinkerArg Include="-ldl" />
       <LinkerArg Include="-lm" />
+      <LinkerArg Include="-lobjc" Condition="'$(TargetOS)' == 'OSX'" />
+      <LinkerArg Include="-lswiftCore" Condition="'$(TargetOS)' == 'OSX'" />
+      <LinkerArg Include="-lswiftFoundation" Condition="'$(TargetOS)' == 'OSX'" />
       <LinkerArg Include="-lz" />
       <LinkerArg Include="-lrt" Condition="'$(TargetOS)' != 'OSX'" />
       <LinkerArg Include="-licucore" Condition="'$(TargetOS)' == 'OSX'" />
+      <LinkerArg Include="-L/usr/lib/swift" Condition="'$(TargetOS)' == 'OSX'" />
       <LinkerArg Include="@(StaticICULibs)" Condition="'$(StaticICULinking)' == 'true'" />
       <LinkerArg Include="-dynamiclib" Condition="'$(TargetOS)' == 'OSX' and '$(NativeLib)' == 'Shared'" />
       <LinkerArg Include="-shared" Condition="'$(TargetOS)' != 'OSX' and '$(NativeLib)' == 'Shared'" />

